src/net.cpp

```diff
@@ -2185,18 +2305,33 @@ uint64_t CNode::GetOutboundTargetBytesLeft()
     return (nMaxOutboundTotalBytesSentInCycle >= nMaxOutboundLimit) ? 0 : nMaxOutboundLimit - nMaxOutboundTotalBytesSentInCycle;
 }
 
-uint64_t CNode::GetTotalBytesRecv()
+uint64_t CConnman::GetTotalBytesRecv()
 {
     LOCK(cs_totalBytesRecv);
     return nTotalBytesRecv;
 }
 
-uint64_t CNode::GetTotalBytesSent()
+uint64_t CConnman::GetTotalBytesSent()
 {
     LOCK(cs_totalBytesSent);
     return nTotalBytesSent;
 }
 
+uint64_t CConnman::GetLocalServices() const
+{
+    return nLocalServices;
+}
+
+void CConnman::SetBestHeight(int height)
```

Comment:
Oh, I see... that issue is not introduced by this PR; what's called GetHeight in main and is not not expected to be non-decreasing is used in net as a "best height". Ensuring modification order for nBestHeight matches the order established by cs_main would be sufficient to maintain the current behaviour.

It does seem like it would be better to have a high-water-mark variable protected by cs_main that main code could use to signal when a new *best* height was reached. I don't see the need for that change necessarily to be included in this PR though.